src/System.Linq/src/System/Linq/Contains.cs

```diff
@@ -26,20 +26,9 @@ public static bool Contains<TSource>(this IEnumerable<TSource> source, TSource v
                 comparer = EqualityComparer<TSource>.Default;
             }
 
-            if (source == null)
-            {
-                throw Error.ArgumentNull(nameof(source));
-            }
-
-            foreach (TSource element in source)
-            {
-                if (comparer.Equals(element, value))
-                {
-                    return true;
-                }
-            }
+            EnumerableHelpers.ThrowIfNull(source, nameof(source));
 
-            return false;
+            return source.Any(element => comparer.Equals(element, value));
```

Comment:
@jamesqo is correct.  This is not a change we can take.  It has a very measurable and negative impact on perf.

I appreciate your contribution, and I agree that removing duplication is in general a good thing, but not when it comes at the expense of performance.  Even the changes to replace the if/throw statements can hurt performance, as at least with the JIT today those helper calls will not be inlined and we'll be adding one or more additional function calls to each operation; for some operations on some inputs, that may not matter, but for others, it very well could.  And without proof that it won't have a negative impact, we'd prefer to keep the slightly larger code.